app/controllers/candidate_interface/decoupled_references/review_controller.rb

```diff
@@ -35,19 +35,23 @@ def submit
       end
 
       def confirm_destroy_referee
-        redirect_to candidate_interface_decoupled_references_review_path unless @reference.not_requested_yet?
+        redirect_to candidate_interface_decoupled_references_review_path and return if @reference.blank?
+        redirect_to candidate_interface_decoupled_references_review_path and return unless @reference.not_requested_yet?
       end
 
       def confirm_destroy_reference
-        redirect_to candidate_interface_decoupled_references_review_path unless @reference.feedback_provided?
+        redirect_to candidate_interface_decoupled_references_review_path and return if @reference.blank?
+        redirect_to candidate_interface_decoupled_references_review_path and return unless @reference.feedback_provided?
```

Comment:
Can we pull the bottom line check up into the if of the top line? (Same for other places too)

Reply:
That makes the boolean expressions a little harder to follow. I've reworked them in a refactor commit - WDYT?